package-res/resources/web/pentaho/util/arg.js

```diff
@@ -20,23 +20,77 @@ define([
 
   var A_slice = Array.prototype.slice;
 
+  /**
+   * @name pentaho.util.arg
+   * @class
+   * @amd pentaho/util/arg
+   *
+   * @classDesc Util methods for keyword args.
+   *
+   * ### AMD
+   *
+   * Module Id: `pentaho/util/arg`
+   *
+   * The AMD module returns the arg utils.
+   *
+   * @description Performs actions on keyword args.
+   */
   return {
+    /**
+     * Provides the optional property value of a given object or the given default value if the given object does
+     * not contain the sought after property value.
+     *
+     * @param {Object} o Keyword arguments.
+     * @param {String} p Property name to find in o param.
+     * @param {Object|String|Number} dv Default value to return if p param is not found.
```

Comment:
I think it would be more readable if we used more legible argument names, e.g. 
@param {Object} keyArgs -  Keyword arguments.
...